frontend/src/pages/Search.js

```diff
@@ -24,75 +24,102 @@ export default function Search() {
       draggable: true,
       progress: undefined,
       theme: "light",
-      });
+    });
   };
 
-  const[isLoading,SetisLoading]=useState(false);
-  const [users,SetUsers]=useState([])
+  const [isLoading, SetisLoading] = useState(false);
+  const [users, SetUsers] = useState([]);
   // const[query,setQuery]=useState('');
-  const[resultsEmpty,setResultsEmpty]=useState(false);
-
+  const [resultsEmpty, setResultsEmpty] = useState(false);
 
-  const onChangeTextHandler=(e)=>{
+  const onChangeTextHandler = (e) => {
     // setQuery(e.target.value);
-    if(!e.target.value)
-    {
+    if (!e.target.value) {
       return;
     }
 
-   const timeout=setTimeout(() => {
-      searchHandler(e.target.value)
+    const timeout = setTimeout(() => {
+      searchHandler(e.target.value);
     }, 1000);
 
-    return ()=>{
+    return () => {
       clearTimeout(timeout);
-    }
-  }
-
-  const searchHandler=async(value)=>{
+    };
+  };
 
+  const searchHandler = async (value) => {
     SetisLoading(true);
-    const cookie=localStorage.getItem('jwt');
-    const response=await fetch(`${process.env.REACT_APP_API_URL}/api/v1/users?search=${value}`,{
-    headers:{
-      'Content-type':'application/json',
-      'Authorization':`Bearer ${cookie}`
-    }
-  })
-  const data=await response.json();
-  SetisLoading(false);
-  data.users.length=data.users.length>2?data.users.length=2:data.users.length;
-  SetUsers(data.users)
-  if(data.users.length===0)
-  setResultsEmpty(true)
-  else
-  setResultsEmpty(false);
-
-  }
-
-    const accessChatHandler=(values)=>{
+    const cookie = localStorage.getItem("jwt");
+    const response = await fetch(
+      `${process.env.REACT_APP_API_URL}/api/v1/users?search=${value}`,
+      {
+        headers: {
+          "Content-type": "application/json",
+          Authorization: `Bearer ${cookie}`,
+        },
+      }
+    );
+    const data = await response.json();
+    SetisLoading(false);
+    data.users.length =
+      data.users.length > 2 ? (data.users.length = 2) : data.users.length;
+    SetUsers(data.users);
+    if (data.users.length === 0) setResultsEmpty(true);
+    else setResultsEmpty(false);
+  };
 
-      const isPresent=state.find((data)=>{
-        return data.email===values.email
-      });
-      dispatch(AddUser(values,state))
+  const accessChatHandler = (values) => {
```

Comment:
Can we have a better way of handling this ? instead of so many loops ?

Reply:
Hey, @Spiral-Memory, I have actually tried all the possible different ways. I thought this is the most optimal and easy understandable code. If you got any ideas regarding that please mention. I'll look into it.

Reply:
Cool then, if you figure out any better way of better style of writing it, do raise a new PR later